wallet-new/src/Cardano/Wallet/Kernel/DB/HdWallet.hs

```diff
@@ -295,7 +307,9 @@ instance HasPrimKey HdAddress where
     type PrimKey HdAddress = HdAddressId
     primKey = _hdAddressId
 
-type HdRootIxs    = '[]
+-- | An 'HdRoot' can be accessed via its 'Core.Address', which is the
+-- Cardano 'Address' derived by the PublicKey associated with it.
+type HdRootIxs    = '[Core.Address]
```

Comment:
This index is superfluous, it adds nothing on top of the primary key index. Perhaps the confusion stems from the name. How about we rename this to `HdRootAdditionalIxs` (and similarly for the other two)?